### ISSUE TITLE
Setting allow_* options for repo not possible via API

### ISSUE THREAD

**Issue by adamel** [go-gitea/gitea#21591]
### Description

When trying to change any of the allow_* options it does not have any effect. Changing for example `external_tracker` and `has_issues` settings work fine.

```
$ curl -s -X 'PATCH' "https://server/gitea/api/v1/repos/${org}/${repo}?access_token=${token}" \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{ "allow_merge_commits": false, "allow_rebase": false, "allow_rebase_explicit": false, "allow_squash_merge": false }' | jq | grep allow
  "allow_merge_commits": true,
  "allow_rebase": true,
  "allow_rebase_explicit": true,
  "allow_squash_merge": true,
$
```

Can not try to reproduce on demo site as it does not allow API access.

### Gitea Version

1.17.3

### Can you reproduce the bug on the Gitea demo site?

No

### Git Version

2.31.1-2.el9.2

### Operating System

AlmaLinux 9

### How are you running Gitea?

Binary distribution, running directly from command line.

### Database

MySQL

**Comment by jolheiser** [comment]
This should be fixed by https://github.com/go-gitea/gitea/pull/21130

Essentially the problem is you still need to specify `has_pulls` or else the related settings are ignored. (prior to the above fix, which will be in `1.18+`

**Comment by adamel** [comment]
Yes, setting `"has_pull_requests": true` in the request works fine as a workaround in 1.17.3, thanks!
